### PR TITLE
Max cappi

### LINKED ISSUE #597: Constant Altitude Plan Position Indicator (CAPPI)
It would be convenient to create a griddisplay class in line with the radardisplay class to be able to plot CAPPIs in km units.

In addition it may be convenient to be able to plot gridmapdisplay and radarmapdisplay in km units.

### PR DESCRIPTION
**Pull request by syedhamidali**
- [x] Closes #597
- [x] Tests added
- [x] Examples Added
- [ ] Documentation reflects changes: Not sure

So, I have added the plot_maxcappi function to the gridmapdisplay, here is what it is supposed to do

```python
import pyart

grid = pyart.io.read_grid("GRID_KGRR20240119_163853_V06.nc")
display = pyart.graph.GridMapDisplay(grid)

display.plot_maxcappi('reflectivity', vmin=0, vmax=60, add_slogan=True)
```
image

### CODE REVIEW

**Comment by syedhamidali** [comment]
@mgrover1 Could you please take a look at these conflicts? I'm not quite sure how to resolve them.

**Comment by mgrover1** [comment]
Sure! I can take a look here - thanks for putting this together 👍

**Comment by mgrover1** [comment]
@syedhamidali - this should resolve the merge conflicts 👍

**Comment by syedhamidali** [comment]
@mgrover1 Thank you!

**Comment by syedhamidali** [comment]
@mgrover1 I'm unsure if these CI errors relate to my commits. Could you help clarify?

File: pyart/core/grid.py [hunk 1]
```diff
@@ -333,55 +333,129 @@ def to_xarray(self):
...
-            ds.z.attrs = get_metadata("z")
-            ds.y.attrs = get_metadata("y")
-            ds.x.attrs = get_metadata("x")
-
-            ds.z.encoding["_FillValue"] = None
-            ds.lat.encoding["_FillValue"] = None
-            ds.lon.encoding["_FillValue"] = None
-
-            # Grab original radar(s) name and number of radars used to make grid
-            ds.attrs["nradar"] = self.nradar
```
Comment (mgrover1) [inline pyart/core/grid.py:378]: @syedhamidali - you are removing nradar here within the attributes which is causing tests to fail

**Review by mgrover1 — CHANGES_REQUESTED** [review]
Here are a few suggestions - this should help resolving some of the failing tests

**Pull request closed without merging**